### PR TITLE
e2e: vfolder invitation

### PR DESCRIPTION
**Pull request by yomybaby**
**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2658?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2658** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2658?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2644** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2644?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2646** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2646?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: e2e/test-util.ts [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
+  },
+  user: {
+    email: "user@lablup.com",
+    password: "C8qnIo29",
+  },
+  user2: {
+    email: "user2@lablup.com",
+    password: "P7oxTDdz",
+  },
+};
```
Comment (gahyuun) [inline e2e/test-util.ts]: I think you should also add domain admin user

File: e2e/test-util.ts [hunk 2]
```diff
@@ -54,43 +80,75 @@ export async function navigateTo(page: Page, path: string) {
...
-  await page.getByLabel('delete_forever').click();
-  await page.getByRole('textbox', { name: 'Type folder name to delete' }).fill(folderName);
   await page.waitForTimeout(1000);
-  await page.getByRole('textbox', { name: 'Type folder name to delete' }).click();
-  
-  await expect(page.locator('#trash-bin-folder-storage').getByText('e2e-test-folder', { exact: true })).toBeVisible();
-  
-  await page.getByRole('button', { name: 'Delete forever' }).click();
-  await expect(page.locator('#trash-bin-folder-storage').getByText('e2e-test-folder', { exact: true })).toBeHidden();
+  await page.locator("vaadin-grid-cell-content").filter({ hasText: folderName }).locator("//following-sibling::*[7]").locator(':visible').getByLabel('delete_forever').click();
```
Comment (gahyuun) [inline e2e/test-util.ts]: I have verified the test is successful, but the code is intermittently successful. It seems to be caused by not getting the updated folder list.

Comment (yomybaby) [reply e2e/test-util.ts]: Please check again. I use 'waitFor` before `fill` a input.

Comment (gahyuun) [reply e2e/test-util.ts]: Test still fails intermittently when running that code in VFolder sharing test

File: e2e/test-util.ts [hunk 3]
```diff
@@ -54,43 +89,74 @@ export async function navigateTo(page: Page, path: string) {
...
   await page.waitForTimeout(1000);
-  await page.getByRole('button', { name: 'delete' }).first().click();
-  await page.locator('#delete-without-confirm-button').getByLabel('delete').click();
-  await page.getByRole('tab', { name: 'delete' }).click();
+  await page.getByRole("button", { name: "delete" }).first().click();
+  await page
+    .locator("#delete-without-confirm-button")
+    .getByLabel("delete")
+    .click();
+  await page.getByRole("tab", { name: "delete" }).click();
```
Comment (gahyuun) [inline e2e/test-util.ts]:
```suggestion
   await page
    .locator("#delete-without-confirm-button")
    .getByLabel("delete")
    .click();
  await page.waitForTimeout(3000);
  await page.getByRole("tab", { name: "delete" }).click();

```

File: e2e/test-util.ts [hunk 4]
```diff
@@ -54,43 +89,74 @@ export async function navigateTo(page: Page, path: string) {
...
-  
-  await expect(page.locator('#trash-bin-folder-storage').getByText('e2e-test-folder', { exact: true })).toBeVisible();
-  
-  await page.getByRole('button', { name: 'Delete forever' }).click();
-  await expect(page.locator('#trash-bin-folder-storage').getByText('e2e-test-folder', { exact: true })).toBeHidden();
+  await page.locator("vaadin-grid-cell-content").filter({ hasText: folderName }).locator("//following-sibling::*[7]").locator(':visible').getByLabel('delete_forever').click();
+  await page
+    .getByRole("textbox", { name: "Type folder name to delete" })
+    .fill(folderName);
+  await page.waitForTimeout(1000);
```
Comment (gahyuun) [inline e2e/test-util.ts]:
```suggestion
```

File: e2e/test-util.ts [hunk 5]
```diff
@@ -54,43 +89,74 @@ export async function navigateTo(page: Page, path: string) {
...
+    .locator("#trash-bin-folder-storage vaadin-grid-cell-content")
+    .filter({ hasText: "Name" })
+    .locator("vaadin-text-field")
+    .nth(1)
+    .locator("input");
+  await nameInputInTrash.waitFor({ state: 'visible' });
   await nameInputInTrash.fill(folderName);
-  await page.getByLabel('delete_forever').click();
-  await page.getByRole('textbox', { name: 'Type folder name to delete' }).fill(folderName);
   await page.waitForTimeout(1000);
```
Comment (gahyuun) [inline e2e/test-util.ts:148]:
```suggestion
```

File: e2e/test-util.ts [hunk 6]
```diff
@@ -54,43 +89,74 @@ export async function navigateTo(page: Page, path: string) {
...
+    .locator("vaadin-text-field")
+    .nth(1)
+    .locator("input");
+  await nameInput.click();
+  await nameInput.fill(inputValue);
+}
+
 export async function createVFolderAndVerify(page: Page, folderName: string) {
-  await navigateTo(page, 'data');
+  await navigateTo(page, "data");
```
Comment (gahyuun) [inline e2e/test-util.ts:101]: Why was navigateTo used?

**Review by gahyuun — CHANGES_REQUESTED** [review]

**Review by ironAiken2 — APPROVED** [review]
LGTM

**Pull request merged**